tests/crypto_policy_test.go

```diff
@@ -196,41 +215,62 @@ func tryToAccessEndpoint(pod core.Pod, subpath string, port uint16, tlsConfig cl
 		subpath = "/" + subpath
 	}
 
-	// ${webhookServiceName}.${deploymentNamespace}.svc is used to access the endpoints we are testing.
-	// It is used here for the metrics as well for simplicity, as it is the CN in the ca_cert
-	// and the metrics just sit on a different port on the same pod.
-	attemptedUrl = fmt.Sprintf("https://%s.%s.svc:%d%s", strategy.GetSSPWebhookServiceName(), strategy.GetSSPDeploymentNameSpace(), port, subpath)
+	// ${serviceName}.${serviceNamespace}.svc is used to access the endpoints we are testing.
+	attemptedUrl = fmt.Sprintf("https://%s.%s.svc:%d%s", serviceName, pod.Namespace, port, subpath)
 	_, err = httpClient.Get(attemptedUrl)
 	return attemptedUrl, err
 }
 
-func (c tlsConfigTestPermutation) testTLSEndpointAccessible(pod core.Pod, subpath string, port uint16, tlsConfig clientTLSOptions) {
-	_, err := tryToAccessEndpoint(pod, subpath, port, tlsConfig)
-	Expect(err).ToNot(HaveOccurred(), "Can't access pod %s, at port %d, with tlsConfig %#v", pod.Name, port, tlsConfig)
```

Comment:
Not sure I understand this change. Why not keep the Expects? Looks more concise to me.

Reply:
Because the next commit calls this function in an `Eventually` loop. The `Expect()` would fail the test immediately.